src/webgpu/web_platform/external_texture/video.spec.ts

```diff
@@ -3,13 +3,241 @@ Tests for external textures from HTMLVideoElement (and other video-type sources?
+      // GPUExternalTexture will be destroyed via microtask upon requestAnimationFrame.
+      const ensureDestroyed = function () {
+        const commandEncoder2 = t.device.createCommandEncoder();
+        const passEncoder2 = commandEncoder2.beginRenderPass({
+          colorAttachments: [
+            {
+              view: colorAttachment.createView(),
+              loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+              storeOp: 'store',
+            },
+          ],
+        });
+        passEncoder2.setPipeline(pipeline);
+        passEncoder2.setBindGroup(0, bindGroup);
+        passEncoder2.draw(6);
+        passEncoder2.endPass();
+
+        t.expectGPUError('validation', () => t.device.queue.submit([commandEncoder2.finish()]));
+      };
+
+      requestAnimationFrame(ensureDestroyed);
+    });
+  });
+
+g.test('importExternalTexture,compute')
+  .desc(
+    `
+Tests that we can import an HTMLVideoElement into a GPUExternalTexture and use it in a compute shader.
+`
+  )
+  .fn(async t => {
+    const videoUrl = getResourcePath('red-green.webmvp8.webm');
+    const video = document.createElement('video');
+    video.src = videoUrl;
+
+    await startPlayingAndWaitForVideo(video, () => {
+      const externalTextureDescriptor = { source: video };
+      const externalTexture = t.device.importExternalTexture(externalTextureDescriptor);
+
+      const result = t.device.createBuffer({
+        size: 8,
+        usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.STORAGE,
+      });
+
+      const pipeline = t.device.createComputePipeline({
+        compute: {
+          module: t.device.createShaderModule({
+            code: `
+              [[block]] struct Data {
+                  height : i32;
+                  width : i32;
+              };
+    
+              [[group(0), binding(0)]] var t : texture_external;
+              [[group(0), binding(1)]] var<storage, read_write> dst : Data;
+    
+              [[stage(compute), workgroup_size(1)]] fn main() {
+                var dims : vec2<i32> = textureDimensions(t);
+                dst.width = dims.x;
+                dst.height = dims.y;
+                return;
+              }
+            `,
+          }),
+          entryPoint: 'main',
+        },
+      });
+
+      const bg = t.device.createBindGroup({
+        entries: [
+          { binding: 0, resource: externalTexture },
+          { binding: 1, resource: { buffer: result, offset: 0, size: 8 } },
+        ],
+        layout: pipeline.getBindGroupLayout(0),
+      });
+
+      const encoder = t.device.createCommandEncoder();
+      const pass = encoder.beginComputePass();
+      pass.setPipeline(pipeline);
+      pass.setBindGroup(0, bg);
+      pass.dispatch(1);
+      pass.endPass();
+      t.device.queue.submit([encoder.finish()]);
+
+      // Check that video dimension are returned correctly as 80x128
+      const resultData = new Uint32Array([0x0000005000000080]);
```

Comment:
This number is too big for uint32array, is this meant to be:
```suggestion
      const resultData = new Uint32Array([80, 128]);
```
?

Reply:
Done.